sim/paladin/righteous_vengeance.go

```diff
@@ -0,0 +1,72 @@
+package paladin
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+const RVTicks = 4
+
+func (paladin *Paladin) registerRV() {
+	if !paladin.hasRune(proto.PaladinRune_RuneCloakRighteousVengeance) {
+		return
+	}
+
+	paladin.RegisterAura(core.Aura{
+		Label:    "Righteous Vengeance",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+
+			if spell.Flags.Matches(SpellFlag_RV) && result.DidCrit() {
+				paladin.procRV(sim, result)
+			}
+		},
+	})
+
+	paladin.rv = paladin.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 440675},
+		SpellSchool: core.SpellSchoolHoly,
+		//DefenseType: core.DefenseTypeMagic,
+		ProcMask:    core.ProcMaskEmpty,  //Not sure
+		Flags:       core.SpellFlagPureDot | core.SpellFlagIgnoreModifiers | core.SpellFlagIgnoreResists,
+
+		DamageMultiplier: 1,
+		ThreatMultiplier: 1,
+
+		Cast: core.CastConfig{
+			IgnoreHaste: true,
+		},
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "Righteous Vengeance",
+			},
+			NumberOfTicks: RVTicks,
+			TickLength:    time.Second * 2,
+			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeTickCounted)
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			spell.SpellMetrics[target.UnitIndex].Hits++
+			spell.Dot(target).ApplyOrReset(sim)
+		},
+	})
+}
+
+func (paladin *Paladin) procRV(sim *core.Simulation, result *core.SpellResult) {
+	dot := paladin.rv.Dot(result.Target)
+
+	newDamage := result.Damage * 0.5
+	outstandingDamage := core.TernaryFloat64(dot.IsActive(), dot.SnapshotBaseDamage*float64(dot.NumberOfTicks-dot.TickCount), 0)
+
+	dot.Snapshot(result.Target, (outstandingDamage+newDamage)/float64(RVTicks), false)
```

Comment:
Do we have confirmation that it works like ignite?

Reply:
Seems to match ignite behavior to me. https://sod.warcraftlogs.com/reports/8F6JnPajkvZ9YVDG#fight=20&type=damage&source=99&view=events&start=4792427&end=4821193
Look at 33.994 (First Crit) - 101.051 (Last RV tick)

00:33.994 Amereyn Judgement of Martyrdom Testwerk *2604*
00:39.152 Amereyn Crusader Strike Testwerk *1917*
00:48.433	Amereyn Divine Storm Testwerk *1931*
00:53.014	Amereyn Crusader Strike Testwerk *1933*
Total Crit damage 8385

Total Dot Damage 5039 (Including partial resists) , 
![image](https://github.com/wowsims/sod/assets/5923944/adad273d-7318-47b6-8a45-f76cafdbf38c)

Only thing is the modifier flags seem questionable because 5039 is > 0.5 * 8350.
So if anything this is a least optimistic implementation possibly?
No AW used during this period.

Reply:
resolved by remove resits flag and setting defense type